### ISSUE TITLE
About 'X' label 

### ISSUE THREAD

**Issue by kugwzk** [kamalkraj/BERT-NER#1]
In my opinion, you should remove the 'X' label's signal in evaluation, because you add more label than standard dataset, so I can't know very well the F1-score increase because the more label of 'X'. I think the 'X' label is not equal the 'O' label in standard dataset and the BERT paper, but in your code it may be same.

**Comment by kamalkraj** [comment]
Label 'X' is not considered for f1 metrics
https://github.com/kamalkraj/BERT-NER/blob/dba5e7d8612a59e58881ad11204674f549fd8c60/run_ner.py#L515

**Comment by kugwzk** [comment]
But did you use in train?

**Comment by kamalkraj** [comment]
For training I used "X"

**Comment by kugwzk** [comment]
I think add more label in the conll2003 NER standard dataset make it not very comparable for previous works. Could you remove 'X' label during training and get a similar result?

**Comment by kamalkraj** [comment]
If you remove "X" while training or replace "X" with "O" , model performance will drop to ~89 f1 score

**Comment by kugwzk** [comment]
So this is my opinion, use ‘X’ label make it high F1-score, it's not fair. I get the similar result about 91.3 F1 score. And I think BERT origin paper is also remove 'X' label because they use document information to get high F1-score. In short, 'X' label don't have any signal.

**Comment by kamalkraj** [comment]
91.3 without using "X" ??

**Comment by kugwzk** [comment]
Yes. I get the word piece output from BERT model, and then map the first token's vector so I get the same numbers vectors as the standard dataset. And then use a softmax matrix to get the final result. But I only use the BERTModel in pytorch_pretrained_bert.

**Comment by kamalkraj** [comment]
For example 
After extracting features for the sentence below
Jim Hen ##son was a puppet ##eer
You're giving only [Jim , Hen , was , a, puppet] hidden states to linear classification layer ?

**Comment by kugwzk** [comment]
Yes. Because I think the fine tune bert could learn this pattern.

**Comment by kamalkraj** [comment]
I will try this way and let you know

**Comment by kamalkraj** [comment]
@kugwzk 
Can you share your code ?

**Comment by kugwzk** [comment]
I just record the origin word position use a dict in python. For example: [Jim Hen ##son was a puppet ##eer] for [0,1,3,4,5], so I padding the origin word sequence again in the classifier layer. It may be slowly :).

**Comment by shisi2015** [comment]
i think we can add mask to X label when training

**Comment by ereday** [comment]
I agree with @kugwzk on "misusage of X label". @tkukurin In the latest version, are you still using X during training (or evaluation) or have you already removed it as suggested ?

**Comment by kamalkraj** [comment]
@ereday 
Latest version still use `X` , I have code without using`X` label , I need to clean the code a bit. I will try to push code by monday

**Comment by toxic2m** [comment]
Hi @kamalkraj ,

I am Nic from NIVIDIA, thanks for your contribution on this project!
I tried to replace [CLS], X with O directly, and removed [SEP].
I think you supposed to release new code today, have you finished?
Thanks.

**Comment by kamalkraj** [comment]
@toxic2m 
Check out branch https://github.com/kamalkraj/BERT-NER/tree/experiment

**Comment by toxic2m** [comment]
Hi @kamalkraj ,

Actually, I already done this part locally, and I suggest you to map [CLS] and [SEP] to O directly.
Then your FC layer only output real number of classifications, can get better performance.
Thanks.

**Comment by sbmaruf** [comment]
Is there anyone able to reproduce BERT_NER paper results? 
The experiment on the master branch supports the result given in the BERT paper. But as @kugwzk mentioned the problem. Is there anyone able to reproduce results without inferred `X` tags?

**Comment by kugwzk** [comment]
@sbmaruf The result of Conll03 NER reported in the BERT origin paper used document context, which is different from the standard sentence-based evaluation. You can see something about that in here: https://github.com/allenai/allennlp/pull/2067#issuecomment-444180824

**Comment by sbmaruf** [comment]
@kugwzk thanks for your reply. but how to add document level context with NER? 
any idea or code repo?
